src/main/java/org/candlepin/pki/certs/EntitlementPayloadGenerator.java

```diff
@@ -0,0 +1,212 @@
+        }
+
+        String management = product.getAttributeValue(Product.Attributes.MANAGEMENT_ENABLED);
+        if (StringUtils.isNotBlank(management)) {
+            // only included if not the default value of false
+            if (management.equalsIgnoreCase("true") || management.equalsIgnoreCase("1")) {
+                toReturn.setManagement(Boolean.TRUE);
+            }
+        }
+
+        String stackingId = product.getAttributeValue(Product.Attributes.STACKING_ID);
+        if (StringUtils.isNotBlank(stackingId)) {
+            toReturn.setStackingId(stackingId);
+        }
+
+        String virtOnly = pool.getAttributeValue(Product.Attributes.VIRT_ONLY);
+        if (StringUtils.isNotBlank(virtOnly)) {
+            // only included if not the default value of false
+            boolean vo = virtOnly.equalsIgnoreCase("true") ||
+                virtOnly.equalsIgnoreCase("1");
+            if (vo) {
+                toReturn.setVirtOnly(vo);
+            }
+        }
+
+        toReturn.setService(createService(pool));
+
+        String usage = product.getAttributeValue(Product.Attributes.USAGE);
+        if (StringUtils.isNotBlank(usage)) {
+            toReturn.setUsage(usage);
+        }
+
+        String roles = product.getAttributeValue(Product.Attributes.ROLES);
+        if (StringUtils.isNotBlank(roles)) {
+            toReturn.setRoles(Util.toList(roles));
+        }
+
+        String addons = product.getAttributeValue(Product.Attributes.ADDONS);
+        if (StringUtils.isNotBlank(addons)) {
+            toReturn.setAddons(Util.toList(addons));
+        }
+        return toReturn;
+    }
+
+    private Service createService(Pool pool) {
+        if (pool.getProduct().getAttributeValue(Product.Attributes.SUPPORT_LEVEL) == null &&
+            pool.getProduct().getAttributeValue(Product.Attributes.SUPPORT_TYPE) == null) {
+            return null;
+        }
+        Service toReturn = new Service();
+        toReturn.setLevel(pool.getProduct().getAttributeValue(Product.Attributes.SUPPORT_LEVEL));
+        toReturn.setType(pool.getProduct().getAttributeValue(Product.Attributes.SUPPORT_TYPE));
+
+        return toReturn;
+    }
+
+    private Order createOrder(Pool pool) {
+        SimpleDateFormat iso8601DateFormat = Util.getUTCDateFormat();
+        Order toReturn = new Order();
+
+        toReturn.setNumber(pool.getOrderNumber());
+        toReturn.setQuantity(pool.getQuantity());
+        toReturn.setStart(iso8601DateFormat.format(pool.getStartDate()));
+        toReturn.setEnd(iso8601DateFormat.format(pool.getEndDate()));
+
+        if (StringUtils.isNotBlank(pool.getContractNumber())) {
+            toReturn.setContract(pool.getContractNumber());
+        }
+
+        if (StringUtils.isNotBlank(pool.getAccountNumber())) {
+            toReturn.setAccount(pool.getAccountNumber());
+        }
+
+        return toReturn;
+    }
+
+    private org.candlepin.model.dto.Pool createPool(Pool pool) {
```

Comment:
there's a duplicate of this in X509V3ExtensionUtil.java still; lets remove it (I don't think its being used there)